software/webapp/src/main/java/brooklyn/entity/webapp/JavaWebAppSshDriver.java

```diff
@@ -89,11 +89,11 @@ protected String inferRootUrl() {
         if (isProtocolEnabled("https")) {
             Integer port = getHttpsPort();
             checkNotNull(port, "HTTPS_PORT sensors not set; is an acceptable port available?");
-            return String.format("https://%s:%s/", getHostname(), port);
+            return String.format("https://%s:%s/", getSubnetHostname(), port);
```

Comment:
Discovered a problem in EC2 with the naming strategy introduced in 89c126340dcf350a3d887782a5ea79a672fe2f17 -- have fixed it with an additional commit here, and detailed comments.

Now confirmed that subnet hostname in AWS does the right thing.

Merging with the add'l commit but @aledsage please look at b68363e8636a3e0c3cea28a2d39aa448e8d50650 in case you see a better way.  (Bear in mind jclouds wants a `Function<...Image...>` and `Image` doesn't make the cloud provider available!)